### ISSUE TITLE
Implement cancel download in progress in Browse screen

### ISSUE THREAD

**Issue by luizgrp** [google/horologist#678]
### WHAT

In the Browse screen, if the user taps on a playlist that is currently being downloaded, it should prompt the user if they wish to cancel download, and cancel the playlist download if they decide to proceed.

### WHY

In order to align with Figma specs.

### HOW

Implementation should be similar to what's implemented in [UampEntityScreen](https://github.com/google/horologist/blob/2a9759acc2c5665d67dcafae791715628125eb56/media-sample/src/main/java/com/google/android/horologist/mediasample/ui/entity/UampEntityScreen.kt) and [UampEntityScreenViewModel](https://github.com/google/horologist/blob/2a9759acc2c5665d67dcafae791715628125eb56/media-sample/src/main/java/com/google/android/horologist/mediasample/ui/entity/UampEntityScreenViewModel.kt#L92)

**Comment by luizgrp** [comment]
Implementation should be similar to what's implemented in [UampEntityScreen](https://github.com/google/horologist/blob/2a9759acc2c5665d67dcafae791715628125eb56/media-sample/src/main/java/com/google/android/horologist/mediasample/ui/entity/UampEntityScreen.kt) and [UampEntityScreenViewModel](https://github.com/google/horologist/blob/2a9759acc2c5665d67dcafae791715628125eb56/media-sample/src/main/java/com/google/android/horologist/mediasample/ui/entity/UampEntityScreenViewModel.kt#L92)

**Comment by luizgrp** [comment]
Blocked by https://github.com/google/horologist/issues/681

**Comment by yschimke** [comment]
Low priority
